src/back/wgsl/writer.rs

```diff
@@ -173,6 +182,31 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
+    fn get_struct_name(&self, module: &Module, handle: Handle<Type>) -> String {
```

Comment:
can we return `&str` instead of `String`?

Reply:
We have 2 self borrows here. I don't know how we can bit this error if we return `&str` here:

```bash
let name = self.get_struct_name(module, ty);
    |                            ---- immutable borrow occurs here
470 |                 write!(self.out, "{}", name)?;
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                 |
    |                 mutable borrow occurs here
    |                 immutable borrow later used by call
```

Reply:
is it trying to associate the `&Module` lifetime with the others? Let's make it not do this:
```rust
fn get_struct_name<'a>(&'a self, module: &Module, handle: Handle<Type>) -> &'a str {
```

Reply:
Actually, I see what's going on there. Let's make this function insert the name into the output stream instead of returning it. I.e. `put_struct_name(&mut self, ..)`

Reply:
Thank you! Fixed.